lib/index.js

```diff
@@ -35,13 +35,22 @@ function traverse(input, strict) {
  */
 
 function object(obj, strict) {
+  // 'each' utility uses obj.length to check whether the obj is array. To avoid incorrect classification, wrap call to 'each' with rename of obj.length
+  if (obj.constructor !== Array && obj.hasOwnProperty('length')) {
+    obj.lengthNonArray = obj.length;
+    delete obj.length;
+  }
   each(obj, function(key, val) {
     if (isodate.is(val, strict)) {
       obj[key] = isodate.parse(val);
     } else if (type(val) === 'object' || type(val) === 'array') {
       traverse(val, strict);
     }
   });
+  if (obj.lengthNonArray) {
```

Comment:
why not add this `if` inside the `each` iterator function? seems a bit excessive w this current approach.

Reply:
If the `if` is inside the `each` then it will run every iteration but you only need to run it once after the loop so I thought it'd be more efficient this way

Reply:
i guess, just seems odd to temporarily delete a property just to "skip" from being iterated on and then to add it back later. i think it's pretty negligible to consider the complexity for this, id rather favor cleaner/more readable code. The `each` already has 2 `if` statements that is used every iteration. What's another `if`

Reply:
If the `if` was inside the each it could give the impression that there is something inside the `if` block that needs to be run every iteration. I thought that putting the two `if`s outside the `each` showed that they are wrapping the `each` with a simple temporary rename to a  property. If you still think it'd be more readable inside the `each` I'll change it just wanted to make sure you heard all my reasoning first.

Reply:
yeah i see your point but still not really convinced. at this point this is definitely nitpick so its more of a javascript lesson... but id be curious to know what the "best practice" is.... @tsholmes ?

i just feel weird about renaming the property, especially mutating the original parameter passed in `obj`